src/solver/optimisation/adequacy_patch_csr/construct_problem_constraints_LHS.cpp

```diff
@@ -0,0 +1,305 @@
+        {
+            NbInterco
+              = MatriceDesContraintesCouplantes->NombreDInterconnexionsDansLaContrainteCouplante;
+            NombreDeTermes = 0;
+            for (int Index = 0; Index < NbInterco; Index++)
+            {
+                Interco = MatriceDesContraintesCouplantes->NumeroDeLInterconnexion[Index];
+                Poids = MatriceDesContraintesCouplantes->PoidsDeLInterconnexion[Index];
+
+                if (ProblemeHebdo->adequacyPatchRuntimeData.originAreaMode[Interco]
+                      == Data::AdequacyPatch::physicalAreaInsideAdqPatch
+                    && ProblemeHebdo->adequacyPatchRuntimeData.extremityAreaMode[Interco]
+                         == Data::AdequacyPatch::physicalAreaInsideAdqPatch)
+                {
+                    Var = ProblemeHebdo->CorrespondanceVarNativesVarOptim[hour]
+                            ->NumeroDeVariableDeLInterconnexion[Interco];
+
+                    if (Var >= 0)
+                    {
+                        Pi[NombreDeTermes] = Poids;
+                        Colonne[NombreDeTermes] = Var;
+                        NombreDeTermes++;
+                    }
+                }
+            }
+
+            if (NombreDeTermes > 0) // current binding constraint contains an interco type 2<->2
+            {
+                hourlyCsrProblem.numberOfConstraintCsrHourlyBinding[CntCouplante]
+                  = ProblemeAResoudre->NombreDeContraintes;
+
+                NomDeLaContrainte = "bc::hourly::" + std::to_string(hour) + "::"
+                                    + MatriceDesContraintesCouplantes->NomDeLaContrainteCouplante;
+
+                logs.debug() << "C (bc): " << ProblemeAResoudre->NombreDeContraintes << ": "
+                             << NomDeLaContrainte;
+
+                OPT_ChargerLaContrainteDansLaMatriceDesContraintes(
+                  ProblemeAResoudre,
+                  Pi,
+                  Colonne,
+                  NombreDeTermes,
+                  MatriceDesContraintesCouplantes->SensDeLaContrainteCouplante);
+            }
+        }
+    }
+}
+
+void OPT_ConstruireLaMatriceDesContraintesDuProblemeQuadratique_CSR(
+  PROBLEME_HEBDO* ProblemeHebdo,
+  HOURLY_CSR_PROBLEM& hourlyCsrProblem)
+{
+    logs.debug() << "[CSR] constraint list:";
+    double* Pi;
+    int* Colonne;
+    PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre;
+    ProblemeAResoudre = ProblemeHebdo->ProblemeAResoudre;
+
+    Pi = (double*)MemAlloc(ProblemeAResoudre->NombreDeVariables * sizeof(double));
+    Colonne = (int*)MemAlloc(ProblemeAResoudre->NombreDeVariables * sizeof(int));
+
+    ProblemeAResoudre->NombreDeContraintes = 0;
+    ProblemeAResoudre->NombreDeTermesDansLaMatriceDesContraintes = 0;
+
+    setConstraintsOnFlows(ProblemeHebdo, hourlyCsrProblem, Pi, Colonne);
+    setNodeBalanceConstraints(ProblemeHebdo, hourlyCsrProblem, Pi, Colonne);
+    setBindingConstraints(ProblemeHebdo, hourlyCsrProblem, Pi, Colonne);
```

Comment:
These 3 previous methods should be private methods of a class **csrQuadraticProblem**